exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.jaeger;
+
+import com.google.protobuf.Duration;
+import com.google.protobuf.Timestamp;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.Model;
+import io.opentelemetry.proto.trace.v1.AttributeValue;
+import io.opentelemetry.proto.trace.v1.Span;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/** Adapts OpenTelemetry objects to Jaeger objects. */
+class Adapter {
+  private Adapter() {}
+
+  /**
+   * Converts a list of {@link Span} into a collection of Jaeger's {@link Model.Span}.
+   *
+   * @param spans the list of spans to be converted
+   * @return the collection of Jaeger spans
+   * @see #toJaeger(Span)
+   */
+  static Collection<Model.Span> toJaeger(List<Span> spans) {
+    List<Model.Span> convertedList = new ArrayList<>(spans.size());
+    for (Span span : spans) {
+      convertedList.add(toJaeger(span));
+    }
+    return convertedList;
+  }
+
+  /**
+   * Converts a single {@link Span} into a Jaeger's {@link Model.Span}.
+   *
+   * @param span the span to be converted
+   * @return the Jaeger span
+   */
+  static Model.Span toJaeger(Span span) {
+    Model.Span.Builder target = Model.Span.newBuilder();
+
+    target.setTraceId(span.getTraceId());
+    target.setSpanId(span.getSpanId());
+    target.setOperationName(span.getName());
+    target.setStartTime(span.getStartTime());
+    target.setDuration(getDuration(span.getStartTime(), span.getEndTime()));
+
+    target.addAllTags(toKeyValues(span.getAttributes()));
+    target.addAllLogs(toJaegerLogs(span.getTimeEvents()));
+    target.addAllReferences(toSpanRefs(span.getLinks()));
```

Comment:
In OpenTelemetry we have a parent Span defined via the getParentSpanId and Links. Every Span in OpenTelemetry has only one parent and could have multiple linked parents.

The links are used usually for async operations, batching operations or when the request comes from a external source and the incoming trace parent is not trusted. Currently the links do not have the ability to distinguish between these cases but it is safe to probably assume FOLLOWS_FROM in Jaeger.

You did miss to add the CHILD_OF reference for the parent SpanId.

/cc @yurishkuro

Reply:
Thanks for the clarification! I did have questions about it (#475), but I think it's clearer now. 

I did miss the `getParentSpanId`, though: good catch!